File: Data Structures & Algorithms/validate-parentheses/submission-0.py
class Solution:
    def isValid(self, s: str) -> bool:
        open_lst = {')':'(', '}':'{', ']':'['}
        stack = deque()
        for c in s:
            if len(stack) != 0 and c in open_lst and stack[-1] == open_lst[c]:
                stack.pop()
            else:
                stack.append(c)
        if len(stack) == 0:
            return True
        return False
                

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-0.cpp
class Solution {
   public:
    int evalRPN(vector<string>& tokens) {
        stack<int> s;
        for (string& it : tokens) {
            if (it == "+" || it == "-" || it == "*" || it == "/") {
                int op2 = s.top();
                s.pop();
                int op1 = s.top();
                s.pop();

                if (it == "+")
                    s.push(op1 + op2);
                else if (it == "-")
                    s.push(op1 - op2);
                else if (it == "*")
                    s.push(op1 * op2);
                else if (it == "/")
                    s.push(op1 / op2);
            } else {
                s.push(stoi(it));
            }
        }
        return s.top();
    }
};
